src/cmdhandler/command_handler.cpp

```diff
@@ -370,28 +440,218 @@ std::string command_handler::collector_system(const std::vector<std::string>& to
+        for (int row = 0; row < camera_matrix.rows; ++row) {
+            for (int col = 0; col < camera_matrix.cols; ++col) {
+                response << camera_matrix.at<double>(row, col);
+                response << ",";
+            }
+        }
+
+        //add distortion_matrix variable
+        response << "\n    distortion_matrix: ";
+        cv::Mat distortion_matrix = current_state.camera.distortion_matrix;
+        for (int row = 0; row < distortion_matrix.rows; ++row) {
+            for (int col = 0; col < distortion_matrix.cols; ++col) {
+                response << distortion_matrix.at<double>(row, col);
+                response << ",";
+            }
+        }
+
+        //add marker_dictionary variable
+        response << "\n    marker_dictionary: ";
+        response << current_state.camera.marker_dictionary;
+
+        //add camera_option variable
+        response << "\n    camera_options: ";
+        for(auto const &option : current_state.camera.camera_options){
+            response << "\n        " << option.first << ": " << std::boolalpha << option.second;
+        }
+
+        return response.str();
+    }else if(tokens[0] == "set"){
+        if(tokens.size() < 3){
+            return "please provide a variable to set\n    ex: set camera url http://example.com";
+        }
+
+        std::string variable = tokens[2];
+
+        if(variable == "url"){
+            if(tokens.size() != 4){
+                return "please provide a value for variable '"+variable+"'\n    ex: set camera url http://example.com";
+            }
+            current_state.camera.url = tokens[3];
+            return "camera url set to '"+tokens[3]+"'";
+        }else if(variable == "camera_matrix" || variable == "distortion_matrix"){
+            // since camera_matrix/distortion_matrix are the same data type/format, just use a conditional to assign a
+            // cv::Mat to the chosen variable
+            if(tokens.size() != 4){
+                return "please provide a comma separated list of 9 doubles for variable '"+variable+"'\n    ex: set camera "+variable+" 1926.56, 0, 1380.786, 0, 1895.6, 745.4, 0, 0, 1";
+            }
+
+            std::vector<std::string> values = tokenize_values_by_commas(tokens[3]);
+
+            if(values.size() != 9){
+                return "please provide a comma separated list of 9 doubles, "+std::to_string(values.size())+" given";
+            }
+
+            try{
+                if(variable == "camera_matrix"){
+                    current_state.camera.camera_matrix = values_by_comma_to_mat(values);
+                }else if(variable == "distortion_matrix"){
+                    current_state.camera.distortion_matrix = values_by_comma_to_mat(values);
+                }
+
+                return "'"+variable+"' variable set with values "+tokens[3];
+            }catch(const std::invalid_argument& err){
+                return "please provide a comma separated list of doubles";
```

Comment:
We probably shouldn't send the exception message back to the client, but it would be good to at least log it so that we can know what exactly went wrong while processing the input